### PR TITLE
Target / Segment Caching 

### PR DESCRIPTION
**Pull request by erdirowlands**
This PR implements a Feature / Segment cache with functionality to pull from the `ff-server` API and store to cache. This functionality is not yet integrated into the `Client` startup flow (where all config is pulled on startup) - so I have tested locally by invoking the relevant functions in a rebar shell. 

What's missing?

- Wiring up of this functionality in a `Client` module. 
- Exception handling.
- Logging.
- Unit tests for the `retrieve` module. 

Linked tickets are: 

1. [FFM-4277 - Setup Cache](https://harness.atlassian.net/browse/FFM-4277)  
2. [FFM-4278 - Retrieve Flag and Target config from Cloud](https://harness.atlassian.net/browse/FFM-4278)

### CODE REVIEW

File: src/cfclient_retrieve.erl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+%% TODO - this type should be part of the Client module when it's created. This isn't Golang
+%% where consumers get to say what the interface should be.
+-type client_config() :: {EnvironmentID :: binary(), BearerToken :: binary(), ClusterID :: binary()}.
+
+%% @doc Retrieve all features from the FF API and store them to cache.
+%% @end
+-spec retrieve_flags(Context :: ctx:t(), CachePID :: pid(), ClientConfig :: client_config()) -> ok | not_ok.
+retrieve_flags(Context, CachePID, ClientConfig) ->
+  {BearerToken, EnvironmentID, ClusterID} = ClientConfig,
+  Optional = #{ cfg => #{auth => #{ 'BearerAuth' => <<"Bearer ", BearerToken/binary>>}, host => "https://config.ff.harness.io"},  params => #{cluster => ClusterID }},
```
Comment (bmjen) [inline src/cfclient_retrieve.erl:22]: Can you add a comment/note to remove the hardcoding when we get the Config handler done?

File: src/cfclient_retrieve.erl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    {ok, Features, Headers} ->
+      [cfclient_cache_repository:set_to_cache({flag, maps:get(feature, Feature)}, Feature, CachePID) || Feature <- Features]
+  end.
+
+%% @doc Retrieve all segments from the FF API and store them to cache.
+%% @end
+-spec retrieve_segments(Context :: ctx:t(), CachePID :: pid(), ClientConfig :: client_config()) -> ok | not_ok.
+retrieve_segments(Context, CachePID, ClientConfig) ->
+  {BearerToken, EnvironmentID, ClusterID} = ClientConfig,
+  Optional = #{ cfg => #{auth => #{ 'BearerAuth' => <<"Bearer ", BearerToken/binary>>}, host => "https://config.ff.harness.io"},  params => #{cluster => ClusterID }},
```
Comment (bmjen) [inline src/cfclient_retrieve.erl:37]: Here too

**Review by bmjen — CHANGES_REQUESTED** [review]

File: src/cfclient_retrieve.erl (hunk 1 above)
Comment (erdirowlands) [reply src/cfclient_retrieve.erl:22]: Done!

File: src/cfclient_retrieve.erl (hunk 2 above)
Comment (erdirowlands) [reply src/cfclient_retrieve.erl:37]: Done

**Pull request merged**